### PR TITLE
Proposal for deployment of default StorageClass

### PR DESCRIPTION
**Pull request by jsafrane**
@kubernetes/sig-storage-proposals

### CODE REVIEW

File: contributors/design-proposals/default-storage-class.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  `cluster/addons/storage-class/<platform>/` - already done in 1.5
+
+* We need a default StorageClass for vSphere, Azure and Photon in `cluster/addons/storage-class/<platform>`
+
+* cluster/kube-up.sh scripts need to be updated to install the storage class on appropriate platforms
+  * Already done on GCE, AWS and OpenStack.
+
+* kops needs to be updated to install the storage class on appropriate platforms (if it is cloud-provider aware)
+
+* kadm needs to be updated to install the storage class on appropriate platforms (if it is cloud-provider aware)
```
Comment (philips) [inline contributors/design-proposals/default-storage-class.md]: what is kadm?

File: contributors/design-proposals/default-storage-class.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* GCE
+* Photon
+* OpenStack
+* vSphere
+
+Explicitly, there is no default storage class on bare metal.
+
+## Motivation
+
+In Kubernetes 1.5, we had "alpha" dynamic provisioning on aforementioned cloud
```
Comment (philips) [inline contributors/design-proposals/default-storage-class.md]: link?

File: contributors/design-proposals/default-storage-class.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+## Goal
+
+Usual Kubernetes installation tools should deploy a default StorageClass
+where it makes sense.
+
+"*Usual installation tools*" are:
+
+* cluster/kube-up.sh
+* kops
+* kadm
```
Comment (philips) [inline contributors/design-proposals/default-storage-class.md]: what is kadm?

Comment (philips) [inline contributors/design-proposals/default-storage-class.md]: cc @aaronlevy @derekparker from bootkube.

**Comment by philips** [comment]
cc @lpabon

File: contributors/design-proposals/default-storage-class.md (hunk 3 above)
Comment (jsafrane) [inline contributors/design-proposals/default-storage-class.md]: sorry, it should be kubeadm. Fixed on both places.

File: contributors/design-proposals/default-storage-class.md (hunk 2 above)
Comment (jsafrane) [inline contributors/design-proposals/default-storage-class.md]: design: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/persistent-storage.md (Kubernetes 1.2 or earlier?), current code: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/plugins.go#L144

**Comment by jsafrane** [comment]
* I talked to kops guys, they support AWS only and they do deploy a default  storage class there.
* I filled a ticket on kubeadm, no response so far: https://github.com/kubernetes/kubeadm/issues/138

**Comment by jsafrane** [comment]
I'm not sure how /lgtm works in this repo, I'm merging it as it was approved by Saad.

**Comment by jsafrane** [comment]
GKE should ship a default storage class too: https://github.com/kubernetes/kubernetes/issues/41559

File: contributors/design-proposals/default-storage-class.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+"*Where it makes sense*" are:
+
+* AWS
+* Azure
+* GCE
+* Photon
+* OpenStack
+* vSphere
+
+Explicitly, there is no default storage class on bare metal.
```
Comment (r4j4h) [inline contributors/design-proposals/default-storage-class.md]: I think it should specify that there is no **provided** default storage class for bare metal.. e.g. things like [nfs-provisioner](https://github.com/kubernetes-incubator/nfs-provisioner) exists and someone can write their own.

**Pull request merged**